specifyweb/specify/record_merging.py

```diff
@@ -229,23 +233,18 @@ def record_merge_fx(model_name: str, old_model_ids: List[int], new_model_id: int
             # Locking foreign objects in the beginning because another transaction could update records, and we will 
             # then either overwrite or delete that change if we iterate to it much later.
             for obj in foreign_objects:
-                # If it is a dependent field, delete the object instead of updating it.
-                # This is done in order to avoid duplicates
+                # If it is a dependent field, continue
                 if table_name in dependant_table_names:
-                    # Note: need to handle case where deletion throws error because it is referenced my other records
-                    try:
-                        clean_fields_pre_delete(obj)
-                        obj.delete()
-                    except ProtectedError as e:
-                        # NOTE: Handle ProtectedError in the future.
-                        # EXAMPLE: ProtectedError: ("Cannot delete some instances of model 'Address' because they are 
-                        # referenced through protected foreign keys:
-                        # 'Division.address'.", {<Division: Division object (2)>})
-                        raise
                     continue
 
-                # Set new value for the field
-                setattr(obj, field_name_id, new_model_id)
+                try:
+                    # Set new value for the field
+                    setattr(obj, field_name_id, new_model_id)
+                except TypeError as e:
+                    if "Direct assignment to the reverse side of a related set is prohibited" in str(e):
+                        continue
+                    else:
+                        raise
```

Comment:
hmm is this really needed? shouldn't we be skipping these relationships anyways? that is, in the place where you are filtering for relationships to the merge model, exclude to-many (from foreign to merge)?

Reply:
also, possible, cache that result (see functools) -- the relationships to merge, as they won't change at runtime. but not a blocker, the above point is.